### DISCUSSION TITLE
Arduino support through pyFirmata

### DISCUSSION THREAD

**Discussion by roaldarbol** [BrianPugh/belay#166] [Ideas]
Hi Brain! Hope you're well!
Just popping in for a quick idea (am terribly busy writing up my PhD, so won't actually be needing new use-cases for at least another year). So basically I'm trying to plan ahead a bit for what is basically a GUI wrapper for Belay + cameras (once people have refined their Belay scripts themselves that is, this is just to execute them, input/output stuff). 
I was thinking about whether Belay might be able to serve Arduino too somehow and came across pyFirmata. Now, I have no idea whether it would be possible/feasible to attempt to implement it, but I thought I'd just leave the idea here to soak a bit. ;-)
Cheers, Mikkel

**Comment by BrianPugh** [comment]
Hey Mikkel!

Unfortunately, Belay will fundamentally not work with arduino. Firmata (and pyFirmata) basically works by flashing a firmware to a microcontroller and then performs fixed actions based on commands sent over serial. Belay works by issuing plain python commands over the micropython REPL. So here micropython/circuitpython being on the board to interpret all those python commands is critical.

**Reply by roaldarbol** [reply]
Ah, I see, fair enough - worth a shot. 😉
